Add builtin erc7201 (#15968)
ethereum/solidity: docs/cheatsheet.rst

  arbitrary precision and does not wrap around at ``2**256``. Assert that ``k != 0`` starting from version 0.5.0.
- ``mulmod(uint x, uint y, uint k) returns (uint)``: compute ``(x * y) % k`` where the multiplication is performed
  with arbitrary precision and does not wrap around at ``2**256``. Assert that ``k != 0`` starting from version 0.5.0.

> - ``erc7201(string memory id) returns (uint)``: compute the root of the namespace storage according to ERC7201 formula.

Reviewer: We should document that fact that, unlike other builtins here, this one is usable in comptime context.

Reviewer: We should actually document mutability for all builtins. For example:
```solidity
erc7201(string memory id) returns (uint)
```
should really be
```solidity
erc7201(string memory id) pure returns (uint)
```

and within `pure` we should note the distinction between ones that are even stricter, i.e. comptime (usable in layouts and array sizing) or runtime constant (usable in `constant` initializers).

That's something to do in a separate PR though, not just randomly for a single builtin.

Author: Right. I created an issue to track that: #16418.

> - ``erc7201(string memory id) returns (uint)``: compute the root of the namespace storage according to ERC7201 formula. It can be used in compile time context.

Reviewer: ```suggestion
- ``erc7201(string memory id) returns (uint)``: compute base slot of an ``erc7201`` storage namespace.
    Can be used in compile-time context.
```

Reviewer: The version you put in the PR has `can` rather than `Can` (missing capitalization).

Author: Fixed.
